tests/acl/test_src_mac_rewrite.py

```diff
@@ -0,0 +1,705 @@
+
+    # --- Build overlay config JSON ---
+    dut_json = {
+        "NEIGH": {
+            "PortChannel101|201.0.0.101": {
+                "neigh": "00:aa:bb:cc:dd:ee",
+                "family": "IPv4"
+            }
+        },
+        "VXLAN_TUNNEL": {
+            tunnel_name: {"src_ip": dut_vtep}
+        },
+        "VNET": {
+            "Vnet1": {
+                "vni": str(vnet_base),
+                "vxlan_tunnel": tunnel_name,
+                "scope": "default",
+                "peer_list": "",
+                "advertise_prefix": "false",
+                "overlay_dmac": "25:35:45:55:65:75"
+            }
+        },
+        "VNET_ROUTE_TUNNEL": {
+            "Vnet1|150.0.3.1/32": {"endpoint": ptf_vtep}
+        }
+    }
+
+    # Copy overlay config to DUT
+    config_content = json.dumps(dut_json, indent=4)
+    logger.info("Applying comprehensive VXLAN/VNET config:\n%s", config_content)
+
+    duthost.copy(content=config_content, dest="/tmp/config_db_vxlan_vnet.json")
+    duthost.shell("cp /tmp/config_db_vxlan_vnet.json /home/admin/config_db_vxlan_vnet.json")
+    duthost.shell("sonic-cfggen -j /tmp/config_db_vxlan_vnet.json --write-to-db")
+    duthost.shell("config save -y")
+
+    # Clean up temp file
+    duthost.shell("rm /tmp/config_db_vxlan_vnet.json")
+
+    duthost.shell("cp /etc/sonic/config_db.json /home/admin/config_db_vxlan_route_persistent.json")
+
+    config_reload(duthost, safe_reload=True, yang_validate=False)
+
+    time.sleep(20)  # wait for DUT to come up after reload
+
+    ecmp_utils.configure_vxlan_switch(duthost, vxlan_port=4789, dutmac=duthost.facts["router_mac"])
+
+    logger.info("=== VXLAN VNET configuration applied and persisted successfully ===")
+
+
+@pytest.mark.parametrize("scenario_name", ["single_ip_test", "range_test"])
+def test_modify_inner_src_mac_egress(setUp, scenario_name):
+    """
+    Test ACL rule for inner source MAC rewriting with VXLAN/VNET configuration.
+    Tests both single IP (/32) and range (/24) ACL rule matching scenarios.
+    """
+    # Extract test data from setUp fixture
+    duthost = setUp['duthost']
+    ptfadapter = setUp['ptfadapter']
+    scenario = setUp['test_scenarios'][scenario_name]
+
+    ptf_port_1 = setUp['ptf_port_1']
+    ptf_port_2 = setUp['ptf_port_2']
+    bind_ports = setUp['bind_ports']
+    loopback_src_ip = setUp['loopback_src_ip']
+
+    # Extract scenario-specific MAC addresses
+    original_inner_src_mac = scenario['original_mac']
+    first_modified_mac = scenario['first_modified_mac']
+    second_modified_mac = scenario['second_modified_mac']
+
+    # Configuration values
+    vnet_1 = "Vnet1"
+    vxlan_tunnel_name = setUp['vxlan_tunnel_name']
+    outer_src_mac = setUp['outer_src_mac']
+    outer_dst_mac = setUp['outer_dst_mac']
+    RULE_NAME = "rule_1"
+    table_name = ACL_TABLE_NAME
+
+    # Standard values from VXLAN/VNET configuration
+    next_hop_ip = "100.0.1.10"  # PTF VTEP endpoint
```

Comment:
this is redefined many times across the file, can we use a single variable and pass it across the file